spec/models/host_spec.rb

```diff
@@ -477,4 +477,119 @@ def assert_remote_credentials_validated
       expect(host.supports_refresh_network_interfaces?).to be_falsey
     end
   end
+
+  describe "#authentication_check_role" do
+    it "returns smartstate" do
+      host = FactoryGirl.build(:host)
+      expect(host.authentication_check_role).to eq('smartstate')
+    end
+  end
+
+  describe "#validate_power_state" do
+    let(:host) { FactoryGirl.create(:host_vmware_esx,
+                                    :ext_management_system => FactoryGirl.create(:ems_vmware),
+                                    :vmm_vendor            => 'vmware') }
+    context "when host power state equal to pstate" do
+      it "returns nil" do
+        expect(host.validate_power_state('on')).to be_nil
+        expect(host.validate_power_state(['on'])).to be_nil
+      end
+    end
+
+    context "when host power state does not equal to pstate" do
+      it "returns available false" do
+        expect(host.validate_power_state('off')).to eq(:available => false,
+                                                       :message   => "The Host is not powered 'off'")
+        expect(host.validate_power_state(['off'])).to eq(:available => false,
+                                                         :message   => "The Host is not powered [\"off\"]")
+      end
+    end
+  end
+
+  describe "vmotion validation methods" do
+    let(:host) { FactoryGirl.create(:host_vmware_esx,
+                                    :ext_management_system => FactoryGirl.create(:ems_vmware),
+                                    :vmm_vendor            => 'vmware') }
+
+    context "#validate_enable_vmotion" do
+      it "returns available true" do
+        expect(host.validate_enable_vmotion).to eq(:available => true, :message => nil)
+      end
+    end
+
+    context "#validate_disable_vmotion" do
+      it "returns available true" do
+        expect(host.validate_disable_vmotion).to eq(:available => true, :message => nil)
+      end
+    end
+
+    context "#validate_vmotion_enabled?" do
+      it "returns available true" do
+        expect(host.validate_vmotion_enabled?).to eq(:available => true, :message => nil)
+      end
+    end
+  end
+
+  describe "#validate_ipmi" do
+    let(:host) { FactoryGirl.create(:host) }
+    subject    { host.validate_ipmi }
+    before do
+      allow_any_instance_of(Authentication).to receive(:after_authentication_changed)
+    end
+
+    context "host does not have ipmi" do
+      it "returns available false" do
+        expect(subject).to eq(:available => false, :message => "The Host is not configured for IPMI")
+      end
+    end
+
+    context "host has ipmi address and does not have ipmi credentials" do
+      it "returns available false" do
+        host.update_attribute(:ipmi_address, "127.0.0.1")
```

Comment:
For this spec and every one thereafter which uses `update_attribute`, don't use `update_attribute`. Instead, override a new `host` with `let` within this spec's context. 

The reason for this is that you are currently executing an additional database UPDATE query after having already created the record. Overriding let would require only one db write and improve spec performance.